typescript/infra/src/tx/govern-transaction-reader.ts

```diff
@@ -153,6 +169,68 @@ export class GovernTransactionReader {
     };
   }
 
+  private isTimelockControllerTransaction(
+    chain: ChainName,
+    tx: AnnotatedEV5Transaction,
+  ): boolean {
+    return (
+      tx.to !== undefined &&
+      timelocks[chain] !== undefined &&
+      eqAddress(tx.to!, timelocks[chain]!)
+    );
+  }
+
+  private async readTimelockControllerTransaction(
+    chain: ChainName,
+    tx: AnnotatedEV5Transaction,
+  ): Promise<GovernTransaction> {
+    if (!tx.data) {
+      throw new Error('No data in TimelockController transaction');
+    }
+
+    const timelockControllerInterface =
+      TimelockController__factory.createInterface();
+    const decoded = timelockControllerInterface.parseTransaction({
+      data: tx.data,
+      value: tx.value,
+    });
+
+    let insight = '';
+    if (
+      decoded.functionFragment.name ===
+      timelockControllerInterface.functions[
+        'schedule(address,uint256,bytes,bytes32,bytes32,uint256)'
+      ].name
+    ) {
+      const [target, value, data, eta, executor, delay] = decoded.args;
+      insight = `Schedule ${target} to be executed at ${eta} with ${value} ${data}. Executor: ${executor}, Delay: ${delay}`;
+    }
+
+    if (
+      decoded.functionFragment.name ===
+      timelockControllerInterface.functions[
+        'execute(address,uint256,bytes,bytes32,bytes32)'
+      ].name
+    ) {
+      const [target, value, data, executor] = decoded.args;
+      insight = `Execute ${target} with ${value} ${data}. Executor: ${executor}`;
+    }
+
+    if (
+      decoded.functionFragment.name ===
+      timelockControllerInterface.functions['cancel(bytes32)'].name
+    ) {
+      const [id] = decoded.args;
+      insight = `Cancel scheduled transaction ${id}`;
+    }
+
+    return {
+      chain,
+      to: `Timelock Controller (${chain} ${timelocks[chain]})`,
```

Comment:
nit: should we maybe just use `tx.to` instead of `timelocks[chain]`? to make this function potentially usable for future situations where we change the criteria for `isTimelockControllerTransaction` -- e.g. if we ever want it to work for a timelock that's different from the ones we have canonically defined

Reply:
just ran into this